WEB/JS/Upload.js

```diff
@@ -0,0 +1,187 @@
+//this file contains the code to handle the upload
+
+//-- HANDLE FILES -----------------------------------------
+
+//handle picture by displaying thumbnail and adding properties
+const handlePicture = function(files) {
+	
+	var previewDiv = document.getElementById("dropZone")
+	
+	//get and display thumbnails
+	for (var i =0; i< files.length; i++) {
+		var file = files[i]
+	
+		var image = document.createElement("img")
+		image.file = file;
+		image.classList.add("toBeUploaded") //this class will be used to select the file that needs uploading
+		
+		previewDiv.appendChild(image)
+		
+		var reader = new FileReader()
+		reader.onload =( function( Img) { return function (event) {Img.src = event.target.result;}; }  ) (image)
+		reader.readAsDataURL(file)
+		
+	}
+	
+}
+
+
+//function to upload a single file
+function FileUpload(image, file) {
+
+	var xhr = new XMLHttpRequest(); //object to send the data
+	var formData = new FormData();
+	
+	formData.append('picture', file);//add prefix to file
+  /*  I commented out the code to show upload progress; let's first make the upload work 
+  	var self = this;
+  	
+  	//this says that when the upload progress we update the throbber object
+	xhr.upload.addEventListener("progress", function(event) {
+    		if (event.lengthComputable) {
+          		var percentage = Math.round((event.loaded * 100) / event.total);
+          		document.getElementById("progressBar").value=percentage;
+        	}
+    	}, false);
+  	
+  	
+  	//this says that when the upload is complete we set the throbber to 100% and remove it
+	xhr.upload.addEventListener("load", function(e){
+    		self.bar.value=100;
+    		document.getElementById("progressBar").style.display='none'
+    	}, false);
+    */
+    
+    //if something goes wrong
+    xhr.onreadystatechange = function() {
+                if (xhr.readyState == 4 && xhr.status == 200) {
+                    alert(xhr.responseText);
+                }
+    };
+    
+    //here we open the request to upload, the second argument is the .php file to call on the server
+	xhr.open("POST", "file.php", true);
+	xhr.send(formData);
+
+}
+
+
+
+// function which individually sends each picture
+const sendFiles = function () {
+	
+	var images = document.querySelectorAll(".toBeUploaded") //get all pictures
+
+	//upload each picture
+	for (var i =0; i< images.length; i++) {
+		new FileUpload(images[i], images[i].file)
+	
+	}
+
+}
+
+
+//---------------------------------------------------------
+
+//-- SECURITY CHECK ---------------------------------------
+const MaxFileSize = 10485760; //10 MB
+
+//this function checks the file which are being uploaded
+let check = function (files) {
+
+	var regex = /^(image\/)(gif|(x-)?png|p?jpeg)$/i;
+
+	if (files.length === 0) {
+		alert('No file to upload')
+		return false//end the function call
+	}
+				
+	for (var i=0; i < files.length; i++) { //check each uploaded file
```

Comment:
Here, if we drag and drop multiple files which contain a single file that doesn't pass the check, no files are being shown/uploaded onto the `dropZone`.
For example, dragging and dropping one pdf along with 10 images prevents the images to be uploaded. I think the one pdf shouldn't stop the uploading of the 10 images. As in, we should show the error message for the pdf, but should also allow all the other pictures to be uploaded.

Maybe we could have another function, `checkFile` which is being called here to check each file individually and then have the `check` function return an array of booleans. Then, instead of 
`if( !check(files)) return //security check failed` (line 145 in this file)
`handlePicture(files)`
we could create another files array which contains only the allowed files (the ones to which `check` returned true), and then handle this new array instead of `files`.

Same thing for uploading files (line 174 in this file).

Let me know if this sounds good or if you have any questions about this idea.